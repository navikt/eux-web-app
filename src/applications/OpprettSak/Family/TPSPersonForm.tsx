import { Alert, Loader, Search } from '@navikt/ds-react'
import PersonCard from 'applications/OpprettSak/PersonCard/PersonCard'
import { Kodeverk, OldFamilieRelasjon, Person } from 'declarations/types'
import { KodeverkPropType } from 'declarations/types.pt'
import useLocalValidation from 'hooks/useLocalValidation'
import _ from 'lodash'
import { FlexDiv, HorizontalSeparatorDiv, PileDiv, VerticalSeparatorDiv } from '@navikt/hoykontrast'
import PT from 'prop-types'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { TPSPersonFormValidationProps, validateTPSPersonForm } from './validation'

export interface TPSPersonFormProps {
  alertMessage: JSX.Element | string | undefined
  alertType: string | undefined
  alertTypesWatched: Array<string> | undefined
  className?: string
  existingFamilyRelationships: Array<OldFamilieRelasjon>
  onRelationReset: () => void
  onSearchFnr: (sok: any) => void
  onTPSPersonAddedFailure: () => void
  onTPSPersonAddedSuccess: (e: any) => void
  person: Person
  personRelatert: Person | null | undefined
  searchingRelatertPerson: boolean
  rolleList: Array<Kodeverk>
  disableAll?: boolean
}

const TPSPersonForm: React.FC<TPSPersonFormProps> = ({
  alertMessage,
  alertType,
  alertTypesWatched = [],
  existingFamilyRelationships,
  onRelationReset,
  onSearchFnr,
  onTPSPersonAddedFailure,
  onTPSPersonAddedSuccess,
  person,
  personRelatert,
  searchingRelatertPerson,
  rolleList,
  disableAll
}: TPSPersonFormProps): JSX.Element => {
  const { t } = useTranslation()
  const namespace = 'tpspersonform'
  const [_query, setQuery] = useState<string>('')
  const [_personRelatert, setPersonRelatert] = useState<OldFamilieRelasjon | undefined>(undefined)
  const [_tpsperson, setTpsPerson] = useState<OldFamilieRelasjon | undefined>(undefined)
  const [validation, resetValidation, performValidation] = useLocalValidation<TPSPersonFormValidationProps>(validateTPSPersonForm, namespace)

  const sokEtterFnr = () => {
    const valid = performValidation({
      person,
      fnr: _query,
      tpsperson: _tpsperson
    })
    if (valid) {
      setPersonRelatert(undefined)
      setTpsPerson(undefined)
      if (_.isFunction(onSearchFnr)) {
        onSearchFnr(_query)
      }
    }
  }

  useEffect(() => {
    if (personRelatert && !_personRelatert) {
      // Fjern relasjoner array, NOTE! det er kun relasjoner som har rolle.
      const person = _.omit(personRelatert, 'relasjoner')
      const tpsperson = personRelatert?.relasjoner?.find((elem: OldFamilieRelasjon) => elem.fnr === person.fnr)
      setTpsPerson(tpsperson)
      if (!tpsperson) {
        setPersonRelatert(person)
      } else {
        if (onRelationReset) {
          onRelationReset()
        }
        setPersonRelatert(undefined)
      }
    }
  }, [personRelatert])

  const updateQuery = (q: string) => {
    setQuery(q)
    resetValidation(namespace + '-fnr-dnr')
    setPersonRelatert(undefined)
    if (onRelationReset) {
      onRelationReset()
    }
  }

  const conflictingPerson = (): boolean => {
    if (_.find(existingFamilyRelationships, (f) => f.fnr === _personRelatert?.fnr) !== undefined) {
      if (onTPSPersonAddedFailure) {
        onTPSPersonAddedFailure()
      }
      return true
    }
    return false
  }

  const leggTilPersonOgRolle = (person: OldFamilieRelasjon) => {
    if (!conflictingPerson()) {
      setQuery('')
      setPersonRelatert(undefined)
      setTpsPerson(undefined)
      if (onRelationReset) {
        onRelationReset()
      }
      /* Person fra TPS har alltid norsk nasjonalitet. Derfor default til denne. */
      if (onTPSPersonAddedSuccess) {
        onTPSPersonAddedSuccess({
          ...person,
          nasjonalitet: 'NO'
        })
      }
    }
  }

  return (
    <>
      <FlexDiv>
        <PileDiv>
          <Search
            id='TPSPersonForm__input-fnr-or-dnr-id'
            data-testid='TPSPersonForm__input-fnr-or-dnr-id'
            label={t('label:fnr-dnr')}
          /* error={validation[namespace + '-fnr-dnr']?.feilmelding} */
            onChange={updateQuery}
            required
            value={_query}
            disabled={searchingRelatertPerson || disableAll}
          >
            <Search.Button onClick={sokEtterFnr}>
              {t('el:button-search')}
              {searchingRelatertPerson && <Loader />}
            </Search.Button>
          </Search>
          {validation[namespace + '-fnr-dnr']?.feilmelding && (
            <>
              <VerticalSeparatorDiv size='0.5' />
              <span className='navds-error-message navds-error-message--medium'>
                {validation[namespace + '-fnr-dnr']?.feilmelding}
              </span>
            </>
          )}
          {_personRelatert?.adressebeskyttelse &&
            <>
              <VerticalSeparatorDiv size='0.5' />
              <Alert size="small" variant='warning'>
                {t('label:sensitivPerson', {gradering: _personRelatert.adressebeskyttelse})}
              </Alert>
            </>
          }
        </PileDiv>

        <HorizontalSeparatorDiv />
        {_personRelatert && (
          <PersonCard
            person={_personRelatert}
            onAddClick={leggTilPersonOgRolle}
            rolleList={rolleList}
            disableAll={disableAll}
          />
        )}
      </FlexDiv>
      <VerticalSeparatorDiv />
      {_tpsperson && (
        <Alert variant='warning'>
          {t('message:error-relation-already-in-tps')}
        </Alert>
      )}
      {alertMessage && alertType && alertTypesWatched.indexOf(alertType) >= 0 && (
        <Alert variant='error'>
          {alertMessage}
        </Alert>
      )}

    </>
  )
}

TPSPersonForm.propTypes = {
  className: PT.string,
  rolleList: PT.arrayOf(KodeverkPropType.isRequired).isRequired
}

export default TPSPersonForm
