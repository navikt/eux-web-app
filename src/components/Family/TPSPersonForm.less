.c-TPSPersonForm {
  display: flex;
  align-items: flex-end;
}
