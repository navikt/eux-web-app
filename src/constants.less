@import './variabler';

@black: #000000;
@font-color: @black;
