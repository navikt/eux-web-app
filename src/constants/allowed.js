import {IS_DEVELOPMENT} from "constants/environment.ts";

export const ALLOWED_TO_FILL_OUT = IS_DEVELOPMENT ? ["H001", "F001"] : ["H001"]
export const ALLOWED_SAKSHANDLINGER = IS_DEVELOPMENT ?
    ["H001", "X001", "X009", "F002", "F004", "F022", "F023", "F026", "F027", "F016", "R001", "R003", "R004", "Delete_Case"] :
    ["H001", "X001", "X009", "F002", "Delete_Case"]

export const HIDDEN_SAKSHANDLINGER = ["multipleParticipants", "singleParticipant"]
export const ALLOWED_SED_HANDLINGER = ["H002", "F002", "X008", "X010", "X011", "X012", ]
export const ALLOWED_SED_EDIT_AND_UPDATE = ["H001", "H002", "F002", "X008", "X010", "X011", "X012"]
