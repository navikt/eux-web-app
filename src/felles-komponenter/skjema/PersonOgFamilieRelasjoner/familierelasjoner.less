@import '../../../constants';

.familierelasjoner {

  .familierelasjoner__utland {
    margin: 2em 0;
  }

  .familierelasjoner__utland__wrapper {
    .row:not(:last-of-type) {margin-bottom: 1.2em;}
  }

  .familierelasjoner__input {
    margin: 0 1em 0 0;
  }

  .familierelasjoner__detailjer {
    display: flex;
    width: 40%;
    flex-direction: column;
    margin: 0;
  }

  .familierelasjoner__tittel, .familierelasjoner__detalj {
    margin: 0;
    padding: 0;
  }

  .familierelasjoner__tittel {
    font-weight: bold;
  }

  .spesialrelasjon {
    float:right;
  }

  .familierelasjoner__knapp {
    display: flex;
    align-self: center;
    justify-self: flex-end;
    margin-top: 1.8em;

    .familierelasjoner__knapp__label {
      display: flex;
      align-self: center;
    }
  }

  .familierelasjoner__knapp__ikon {
    margin: -0.0em 0.5em 0 0;

    path {
      stroke: @blue-nav;
    }
  }

  .knapp--disabled {
    path {
      stroke: @white !important;
    }
  }

  .familierelasjoner__knapp {

    &:hover:not(.knapp--disabled) {
      .familierelasjoner__knapp__ikon path {
        stroke: @white !important;
      }
    }
  }
}
