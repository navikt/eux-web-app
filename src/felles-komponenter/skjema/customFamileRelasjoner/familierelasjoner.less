@import '../../../constants';

.familerelasjoner {

  .familierelasjoner__linje {
    display: flex;
    flex-direction: row;
    margin: 2em 0;
    width: 800px;
  }

  .familierelasjoner__input {
    margin: 0 1em 0 0;
  }

  .familierelasjoner__detailjer {
    display: flex;
    width: 40%;
    flex-direction: column;
    margin: 0;
  }

  .familierelasjoner__tittel, .familierelasjoner__detalj {
    margin: 0;
    padding: 0;
  }

  .familierelasjoner__tittel {
    font-weight: bold;
  }

  .familierelasjoner__knapp {
    display: flex;
    align-self: flex-start;
    margin-top: 1.9em;
    align-items: center;
  }

  .familierelasjoner__knapp__ikon {
    margin-right: .5em;

    path {
      stroke: @blue-nav;
    }
  }

  .familierelasjoner__knapp {
    &:hover {
      .familierelasjoner__knapp__ikon path {
        stroke: @white !important;
      }
    }
  }
}
