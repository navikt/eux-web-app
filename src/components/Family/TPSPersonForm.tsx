import classNames from 'classnames'
import Alert, { AlertStatus } from '..//Alert/Alert'
import PersonCard from '..//PersonCard/PersonCard'
import { Cell, HorizontalSeparatorDiv, Row, VerticalSeparatorDiv } from '../StyledComponents'
import * as types from '../../constants/actionTypes'
import { FamilieRelasjon, Kodeverk, Person } from '../../declarations/types'
import { KodeverkPropType } from '../../declarations/types.pt'
import _ from 'lodash'
import AlertStripe from 'nav-frontend-alertstriper'
import { Knapp } from 'nav-frontend-knapper'
import { Input } from 'nav-frontend-skjema'
import PT from 'prop-types'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'

export interface TPSPersonFormProps {
  alertStatus: string | undefined
  alertMessage: string | undefined
  alertType: string | undefined
  className?: string
  onAddFailure: () => void
  onAlertClose: () => void
  onAddSuccess: (e: any) => void
  onResetPersonRelatert: () => void
  personRelatert: Person | undefined
  person: Person
  rolleList: Array<Kodeverk>
  onSearchFnr: (sok: any) => void
  existingFamilyRelationships: Array<FamilieRelasjon>
}

const Container = styled.div`
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
  display: flex;
  flex-direction: column;
`
const AlertstripeDiv = styled.div`
  margin: 0.5rem;
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
  width: 50%;
`
const MarginDiv = styled.div`
  margin: 0.5rem;
`
const AlignCenterCell = styled(Cell)`
  display: flex;
  align-items: center;
`
const TPSPersonForm: React.FC<TPSPersonFormProps> = ({
  alertStatus,
  alertMessage,
  alertType,
  className,
  onAddFailure,
  onAddSuccess,
  onAlertClose,
  onResetPersonRelatert,
  personRelatert,
  person,
  rolleList,
  onSearchFnr,
  existingFamilyRelationships
}: TPSPersonFormProps): JSX.Element => {
  const [sok, setSok] = useState('')
  const [_personRelatert, setPersonRelatert] = useState<
    FamilieRelasjon | undefined
  >(undefined)
  const [tpsperson, setTpsPerson] = useState<FamilieRelasjon | undefined>(
    undefined
  )

  const { t } = useTranslation()

  const sokEtterFnr = () => {
    setPersonRelatert(undefined)
    setTpsPerson(undefined)
    if (_.isFunction(onSearchFnr)) {
      onSearchFnr(sok)
    }
  }

  useEffect(() => {
    if (personRelatert && !_personRelatert) {
      // Fjern relasjoner array, NOTE! det er kun relasjoner som har rolle.
      const person = _.omit(personRelatert, 'relasjoner')
      const tpsperson =
        personRelatert && personRelatert.relasjoner
          ? personRelatert.relasjoner.find(
            (elem: FamilieRelasjon) => elem.fnr === person.fnr
          )
          : undefined
      setTpsPerson(tpsperson)
      if (!tpsperson) {
        setPersonRelatert(person)
      } else {
        if (onResetPersonRelatert) {
          onResetPersonRelatert()
        }
        setPersonRelatert(undefined)
      }
    }
  }, [personRelatert, _personRelatert, onResetPersonRelatert])

  const updateSok = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSok(e.target.value)
    setPersonRelatert(undefined)
    if (onResetPersonRelatert) {
      onResetPersonRelatert()
    }
  }

  const conflictingPerson = (): boolean => {
    const { fnr } = _personRelatert!
    if (
      _.find(existingFamilyRelationships, (f) => f.fnr === fnr) !== undefined
    ) {
      if (onAddFailure) {
        onAddFailure()
      }
      return true
    }
    return false
  }

  const leggTilPersonOgRolle = (person: FamilieRelasjon) => {
    if (!conflictingPerson()) {
      setSok('')
      setPersonRelatert(undefined)
      setTpsPerson(undefined)
      if (onResetPersonRelatert) {
        onResetPersonRelatert()
      }
      /* Person fra TPS har alltid norsk nasjonalitet. Derfor default til denne. */
      if (onAddSuccess) {
        onAddSuccess({
          ...person,
          nasjonalitet: 'NO'
        })
      }
    }
  }

  return (
    <Container>
      <Row
        className={classNames(className, 'slideAnimate', {
          feil: !!alertMessage
        })}
      >
        <Cell>
          <Input
            data-testid='c-TPSPersonForm__input-fnr-or-dnr-id'
            label={t('ui:label-fnr-or-dnr')}
            placeholder={t('ui:label-fnr-or-dnr')}
            value={sok}
            onChange={updateSok}
          />
          <VerticalSeparatorDiv />
        </Cell>
        <HorizontalSeparatorDiv />
        <AlignCenterCell>
          <Knapp
            disabled={person.fnr === sok}
            onClick={sokEtterFnr}
          >
            {t('ui:form-search')}
          </Knapp>
        </AlignCenterCell>
      </Row>
      <VerticalSeparatorDiv />
      <Row>
        <Cell>
          {person.fnr === sok && (
            <AlertstripeDiv>
              <AlertStripe type='advarsel'>
                {t('ui:error-fnr-is-user', { sok: sok })}
              </AlertStripe>
            </AlertstripeDiv>
          )}
          {tpsperson && (
            <AlertstripeDiv>
              <AlertStripe className='mt-4 mb-4' type='advarsel'>
                {t('ui:error-relation-already-in-tps')}
              </AlertStripe>
            </AlertstripeDiv>
          )}
          {alertMessage &&
          (alertType === types.SAK_PERSON_RELATERT_GET_FAILURE ||
            alertType === types.FORM_TPSPERSON_ADD_FAILURE) &&
            (
              <AlertstripeDiv>
                <Alert
                  type='client'
                  fixed={false}
                  message={t(alertMessage)}
                  status={alertStatus as AlertStatus}
                  onClose={onAlertClose}
                />
              </AlertstripeDiv>
            )}
          {_personRelatert &&
            (
              <MarginDiv>
                <PersonCard
                  person={_personRelatert}
                  onAddClick={leggTilPersonOgRolle}
                  rolleList={rolleList}
                />
              </MarginDiv>
            )}
        </Cell>
      </Row>
    </Container>
  )
}

TPSPersonForm.propTypes = {
  className: PT.string,
  rolleList: PT.arrayOf(KodeverkPropType.isRequired).isRequired
}

export default TPSPersonForm
