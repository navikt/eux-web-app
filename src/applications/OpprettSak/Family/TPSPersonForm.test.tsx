import * as types from 'constants/actionTypes'
import { familierelasjoner } from '@navikt/eessi-kodeverk/lib/familierelasjoner'
import { render } from '@testing-library/react'
import React from 'react'
import samplePerson from 'resources/tests/samplePerson'
import TPSPersonForm, { TPSPersonFormProps } from './TPSPersonForm'

describe('components/Family/TPSPersonForm', () => {
  let wrapper: any
  const initialMockProps: TPSPersonFormProps = {
    alertMessage: 'Message',
    alertType: types.PERSON_RELATERT_SEARCH_FAILURE,
    alertTypesWatched: [],
    existingFamilyRelationships: [],
    onRelationReset: jest.fn(),
    onSearchFnr: jest.fn(),
    onTPSPersonAddedFailure: jest.fn(),
    onTPSPersonAddedSuccess: jest.fn(),
    person: samplePerson,
    searchingRelatertPerson: false,
    personRelatert: undefined,
    rolleList: familierelasjoner
  }

  beforeEach(() => {
    wrapper = render(<TPSPersonForm {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Render: match snapshot', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Render: HTML structure', () => {
    expect(wrapper.exists('.c-TPSPersonForm')).toBeTruthy()
  })
})
