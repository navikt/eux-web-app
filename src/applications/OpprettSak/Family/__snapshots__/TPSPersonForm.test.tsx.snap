// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Family/TPSPersonForm Render: match snapshot 1`] = `
.c6 {
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
  background-color: inherit !important;
  color: !important;
  border-color: !important;
}

.c6:hover:not(:disabled) {
  background-color: !important;
  color: !important;
}

.c6:hover:not(:disabled) svg {
  color: !important;
}

.c6:hover:not(:disabled) svg path {
  stroke: !important;
}

.c6:disabled {
  color: !important;
  background-color: !important;
  border-color: !important;
}

.c6:disabled svg {
  color: !important;
}

.c6:disabled svg path {
  stroke: !important;
}

.c6:disabled .knapp__spinner {
  border: 2px solid !important;
  border-top-color: !important;
  border-right-color: !important;
}

.c6 .knapp__spinner {
  border: 2px solid !important;
  border-left-color: !important;
  border-bottom-color: !important;
}

.c6:not(:disabled) svg {
  color: !important;
}

.c6:not(:disabled) svg path {
  stroke: !important;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -0.5rem;
  margin-right: -0.5rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

.c4 {
  display: inline-block;
  margin-left: 1rem;
}

.c3 {
  margin-bottom: 1rem;
}

.c0 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:768px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<TPSPersonForm
  alertMessage="Message"
  alertStatus="OK"
  alertType="SAK/PERSON/RELATERT/GET/FAILURE"
  alertTypesWatched={Array []}
  existingFamilyRelationships={Array []}
  onAlertClose={[MockFunction]}
  onRelationReset={[MockFunction]}
  onSearchFnr={[MockFunction]}
  onTPSPersonAddedFailure={[MockFunction]}
  onTPSPersonAddedSuccess={[MockFunction]}
  person={
    Object {
      "etternavn": "BLYANT",
      "fdato": "1961-02-02",
      "fnr": "02026100715",
      "fornavn": "STOR",
      "kjoenn": "M",
      "relasjoner": Array [
        Object {
          "etternavn": "BLYANT",
          "fdato": "1961-04-08",
          "fnr": "02026100714",
          "fornavn": "PEN",
          "kjoenn": "K",
          "rolle": "EKTE",
        },
        Object {
          "etternavn": "BLYANT",
          "fdato": "1980-02-02",
          "fnr": "02026100713",
          "fornavn": "LITEN",
          "kjoenn": "M",
          "rolle": "BARN",
        },
      ],
    }
  }
  rolleList={
    Array [
      Object {
        "kode": "BARN",
        "term": "Barn av",
      },
      Object {
        "kode": "EKTE",
        "term": "Ektefelle til",
      },
      Object {
        "kode": "REPA",
        "term": "Registrert partner med",
      },
      Object {
        "kode": "SAMB",
        "term": "Samboer med",
      },
      Object {
        "kode": "ANNEN",
        "term": "Annen person",
      },
    ]
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div
        className="slideInFromLeft feil"
      >
        <div
          className="c1 slideInFromLeft feil"
        >
          <styled.div>
            <div
              className="c2"
            >
              <Input
                bredde="fullbredde"
                data-test-id="TPSPersonForm__input-fnr-or-dnr-id"
                label="label:fnr-dnr"
                onChange={[Function]}
                placeholder="label:fnr-dnr"
                value=""
              >
                <div
                  className="skjemaelement"
                >
                  <Label
                    htmlFor="32645966-04709-06956-10607-2876069867041"
                  >
                    <label
                      className="skjemaelement__label"
                      htmlFor="32645966-04709-06956-10607-2876069867041"
                    >
                      label:fnr-dnr
                    </label>
                  </Label>
                  <input
                    aria-invalid={false}
                    className="skjemaelement__input input--fullbredde"
                    data-test-id="TPSPersonForm__input-fnr-or-dnr-id"
                    id="32645966-04709-06956-10607-2876069867041"
                    onChange={[Function]}
                    placeholder="label:fnr-dnr"
                    type="text"
                    value=""
                  />
                </div>
              </Input>
              <styled.div>
                <div
                  className="c3"
                />
              </styled.div>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c4"
            />
          </styled.div>
          <Styled(styled.div)>
            <div
              className="c2 c5"
            >
              <Styled(u)
                disabled={false}
                onClick={[Function]}
              >
                <u
                  className="c6"
                  disabled={false}
                  onClick={[Function]}
                >
                  <p
                    autoDisableVedSpinner={false}
                    className="c6"
                    disabled={false}
                    htmlType="submit"
                    inaktivKlasseVedDisabled={false}
                    kompakt={false}
                    mini={false}
                    onClick={[Function]}
                    spinner={false}
                    type="standard"
                  >
                    <button
                      className="knapp c6 knapp--standard"
                      disabled={false}
                      onClick={[Function]}
                      type="submit"
                    >
                      el:button-search
                    </button>
                  </p>
                </u>
              </Styled(u)>
            </div>
          </Styled(styled.div)>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="c3"
        />
      </styled.div>
      <styled.div>
        <div
          className="c1"
        >
          <styled.div>
            <div
              className="c2"
            />
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</TPSPersonForm>
`;
